src/User.cpp

```diff
@@ -218,4 +218,32 @@ void User::sendPrivateMessageToUser(const User &destination, const std::string&
     Logger::debug("Sending private message to " + destination.getNickname() + " from " + this->getNickname() + ": " + message);
     std::string response = ":" + this->_nickname + " PRIVMSG " + destination.getNickname() + " :" + message;
     Server::getInstance().sendMessage(destination.getFd(), response);
+}
+
+/**
+ * This function aims to check if the user is an operator of a channel.
+ * 
+ * @param channelName The name of the channel.
+ * 
+ * @return `true` if the user is an operator of the channel, `false` otherwise.
+ */
+bool User::isUserAnOper(std::string channelName) const {
+    std::vector<Channel>::const_iterator it = findChannel(channelName);
+    if (it != this->_channels.end())
+        return it->findOperUser(this->_nickname);
+    return false;
+}
+
+/**
+ * This function aims to find and get a channel by the name.
+ * 
+ * @param channelName The name of the channel.
+ * 
+ * @return The pointer to the channel with the name.
+ */
+Channel User::findAndGetChannel(std::string channelName) {
+    std::vector<Channel>::iterator it = findChannel(channelName);
+    if (it != this->_channels.end())
+        return *it;
+    return Channel("", *this);
```

Comment:
Fíjate que aquí no haces un return new y en otros sitios sí haces el new, no hacer return new Clase(), implica dos cosas:
- El objeto Clase muere antes de salir de la función, porque al no ser creado con new, no va al heap de memoria.
- al ponerlo en el return estamos haciendo "trabajar" al compilador, que se da cuenta de que quieres usar eso fuera de la función y crea ootra copia más del objeto para que puedas usarlo fuera.

Por otro lado meterle el new también implica que hay que mirar con lupa a ver dónde va ese objeto porque luego habrá que hacerle un delete para borrarlo de ese heap...

En este caso entiendo que has puesto un return Channel porque el compilador te exige que devuelvas un canal. Si hacemos lo que te comentaba de devolver un puntero, ahora podríamos devolver NULL si no lo encontramos, y nos libramos de tener que crear un objeto nuevo, que habría que ver como liberarlo luego

Reply:
Añado a lo de Ale que la estructura tipo:
```
    if (it != this->_channels.end())
        return *it;
    return Channel("", *this);
```
Se puede simplificar usando el operador ternario `?`. Aquí estaría el cambio para este ejemplo:
```
return (it != this->_channels.end()) ? *it
                                     : Channel("", *this);
```
Obviamente no pondrás `Channel("", *this)`, sino lo que te ha comentado Ale